src/algebra/order/floor.lean

```diff
@@ -68,10 +64,18 @@ instance : floor_semiring ℕ :=
   gc_floor := λ n a ha, by { rw nat.cast_id, refl },
   gc_ceil := λ n a, by { rw nat.cast_id, refl } }
 
+@[priority 100] -- see Note [lower instance priority]
+instance ordered_semiring.nontrivial [ordered_semiring α] [floor_semiring α] : nontrivial α :=
```

Comment:
```suggestion
instance floor_semiring.to_nontrivial [ordered_semiring α] [floor_semiring α] : nontrivial α :=
```

Reply:
Yes exactly. It would be better having it now than later given that it might probably change how you generalize stuff.